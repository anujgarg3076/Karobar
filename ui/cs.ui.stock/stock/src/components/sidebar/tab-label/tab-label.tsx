import { Typography, TypographyProps } from '@mui/material'
import React from 'react'
import { LabelProps } from './tab-label.types'

const TabText = (props: TypographyProps) => {
  return <Typography {...props} sx={{ textTransform: 'none', ...props.sx }} />
}

export const TabLabel = ({ group }: LabelProps) => {
  const tabHtml = group.statistics.map((oneStat) => (
    <div key={oneStat.name}>
      {/* This is separated out into 3 elements for easier automated testing */}
      <TabText variant="caption">{oneStat.name}</TabText>
      <TabText variant="caption">: </TabText>
      <TabText variant="caption">{oneStat.value}</TabText>
    </div>
  ))
  return (
    <>
      <TabText variant="h6" sx={{ color: 'black' }}>
        {group.name}
      </TabText>
      {tabHtml}
    </>
  )
}
