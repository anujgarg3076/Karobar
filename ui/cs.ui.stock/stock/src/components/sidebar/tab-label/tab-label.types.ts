interface Statistics {
  name: string
  value: number
}
export interface GroupStatistics {
  name: string
  statistics: Statistics[]
}

export interface LabelProps {
  group: GroupStatistics
}
