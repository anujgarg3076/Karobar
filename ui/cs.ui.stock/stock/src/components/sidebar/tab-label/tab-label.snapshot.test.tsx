import React from 'react'
import renderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { TabLabel } from './tab-label'

it('is rendered correctly', () => {
  const statistics = [
    { name: 'Installed', value: 8 },
    { name: 'Replaced', value: 3 },
  ]
  const group = { name: 'POS', statistics }
  const component = renderer.create(<TabLabel group={group} />)
  const tree = component.toJSON() as ReactTestRendererJSON
  expect(tree).toMatchSnapshot()
})
