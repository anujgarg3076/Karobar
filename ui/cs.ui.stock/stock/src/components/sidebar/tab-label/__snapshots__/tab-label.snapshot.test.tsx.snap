// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is rendered correctly 1`] = `
[
  <h6
    className="MuiTypography-root MuiTypography-h6 css-170lecc-MuiTypography-root"
  >
    POS
  </h6>,
  <div>
    <span
      className="MuiTypography-root MuiTypography-caption css-w5jfwh-MuiTypography-root"
    >
      Installed
    </span>
    <span
      className="MuiTypography-root MuiTypography-caption css-w5jfwh-MuiTypography-root"
    >
      : 
    </span>
    <span
      className="MuiTypography-root MuiTypography-caption css-w5jfwh-MuiTypography-root"
    >
      8
    </span>
  </div>,
  <div>
    <span
      className="MuiTypography-root MuiTypography-caption css-w5jfwh-MuiTypography-root"
    >
      Replaced
    </span>
    <span
      className="MuiTypography-root MuiTypography-caption css-w5jfwh-MuiTypography-root"
    >
      : 
    </span>
    <span
      className="MuiTypography-root MuiTypography-caption css-w5jfwh-MuiTypography-root"
    >
      3
    </span>
  </div>,
]
`;
